src/Chainweb/Chainweb/MinerResources.hs

```diff
@@ -64,11 +73,21 @@ withMiningCoordination logger enabled cutDb inner
     | not enabled = inner Nothing
     | otherwise = do
         t <- newTVarIO mempty
-        inner . Just $ MiningCoordination
+        fmap snd . concurrently (prune t) $ inner . Just $ MiningCoordination
             { _coordLogger = logger
             , _coordCutDb = cutDb
-            , _coordState = t
-            }
+            , _coordState = t }
+  where
+    prune :: TVar MiningState -> IO a
+    prune t = do
+        let !d = 600000000  -- 10 minutes
+        threadDelay d
+        ago <- over (_Unwrapped . _Unwrapped) (subtract (int d)) <$> getCurrentTimeIntegral
+        atomically . modifyTVar' t $ over _Unwrapped (HM.filter (f ago))
+        prune t
+
+    f :: Time Micros -> T3 a PrevBlock b -> Bool
+    f ago (T3 _ (PrevBlock p) _) = _blockCreationTime p > BlockCreationTime ago
```

Comment:
Would it make sense to use the block height as a criterium instead? For instance if the block is < current height - x? With x something like 3?

Reply:
An maybe an additional limit on total items. Say, reject new requests (e.g. via 503) if there are more than, say, 1000 item in the map?

Reply:
The first isn't possible, but the second might be in a follow-up.